src/StorageProvider.ts

```diff
@@ -0,0 +1,21 @@
+import type { consentTypes } from './types/index.js'
+
+export interface StorageProviderInterface {
+  setStore: (values: consentTypes[]) => void
+  getStore: () => consentTypes[]
+}
+
+export class StorageProvider {
+  constructor (options: StorageProviderInterface) {
+    this.setStore = options.setStore ?? this.setStore
+    this.getStore = options.getStore ?? this.setStore
```

Comment:
```
if (options.setStore) {
  this.setStore = options.setStore
}
```
otherwise, throwing an error that the method isn't implemented. It allows someone to build a StorageProvider "implementation" without needing to define both. Extending StorageProvider, or implementing it, does not allow for constructing of a StorageProvider that can be passed to MetricsProvider, which means you can't build a storageProvider that has no `setStore` function. 

Mainly for testing, but also for flexibility of the unknown. It's not unsafe and is supporting of various futures.